pkg/aws/compute/ec2/ec2.go

```diff
@@ -0,0 +1,78 @@
+package ec2
+
+import (
+	"context"
+	"log/slog"
+	"time"
+
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/prometheus/client_golang/prometheus"
+
+	ec2client "github.com/grafana/cloudcost-exporter/pkg/aws/services/ec2"
+	pricingClient "github.com/grafana/cloudcost-exporter/pkg/aws/services/pricing"
+	"github.com/grafana/cloudcost-exporter/pkg/provider"
+)
+
+const (
+	subsystem = "aws_ec2"
+)
+
+// Collector is a prometheus collector that collects metrics from AWS EKS clusters.
+type Collector struct {
+	Region          string
+	Regions         []ec2Types.Region
+	Profile         string
+	Profiles        []string
+	ScrapeInterval  time.Duration
+	pricingService  pricingClient.Pricing
+	ec2Client       ec2client.EC2
+	NextScrape      time.Time
+	ec2RegionClient map[string]ec2client.EC2
+	logger          *slog.Logger
+	context         context.Context
+}
+
+type Config struct {
+	Regions []ec2Types.Region
+	Logger  *slog.Logger
+}
+
+// CollectMetrics is a no-op function that satisfies the provider.Collector interface.
+// Deprecated: CollectMetrics is deprecated and will be removed in a future release.
+func (c *Collector) CollectMetrics(_ chan<- prometheus.Metric) float64 {
+	return 0
+}
+
+// Collect satisfies the provider.Collector interface.
+func (c *Collector) Collect(_ chan<- prometheus.Metric) error {
+	c.logger.LogAttrs(c.context, slog.LevelInfo, "TODO - Implement ec2.Describe")
+	return nil
+}
+
+func (c *Collector) Describe(_ chan<- *prometheus.Desc) error {
+	c.logger.LogAttrs(c.context, slog.LevelInfo, "TODO - Implement ec2.Describe")
+	return nil
+}
+
+func (c *Collector) Name() string {
+	return subsystem
+}
+
+// New creates an AWS EC2 collector.
+func New(ctx context.Context, config *Config, ps pricingClient.Pricing, ec2s ec2client.EC2, regionClientMap map[string]ec2client.EC2) *Collector {
+	logger := config.Logger.With("collector", "ec2")
```

Comment:
SUPER NIT - should this be 

```suggestion
	logger := config.Logger.With("collector", subsystem)

```

Reply:
Only reason why I didn't is because then I think it becomes redundant. The logs would go from

```
time=2024-06-25T06:17:58.286-04:00 level=INFO msg="Registering AWS EC2 collector" provider=aws collector=ec2
```

to 

```
time=2024-06-25T06:18:33.250-04:00 level=INFO msg="Registering AWS EC2 collector" provider=aws collector=aws_ec2
```

Reply:
I think we'll need to rethink a bit the `sub_system` const and how it's being used. In EKS it's used for the metric name, which makes total sense: https://github.com/grafana/cloudcost-exporter/blob/7e496948ea16859fa5bd8bbcd45b911243a34344/pkg/aws/compute/eks/eks.go#L42

That translates to a metric named `cloudcost_aws_eks_instance_cpu_usd_per_core_hour` which aligns with our metric naming convention. In log world I think that's duplicative, but maybe that's fine.